pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -519,19 +510,27 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 				"accepted", isAccepted(pr),
 				"bookingExpired", isBookingExpired(pr),
 				"capacityRevoked", isCapacityRevoked(pr))
+			retryStrategy := ptr.Deref(prc.Spec.RetryStrategy, kueue.DefaultRetryStrategy)
 			switch {
 			case isFailed(pr):
-				if checkState.State != kueue.CheckStateRejected {
-					if attempt := getAttempt(log, pr, wl.Name, check); attempt <= c.maxRetries {
-						// it is going to be retried
-						message := fmt.Sprintf("Retrying after failure: %s", apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message)
+				if checkState.State == kueue.CheckStateRejected {
```

Comment:
Maybe include as part of the prep PR: I suggested there: https://github.com/kubernetes-sigs/kueue/pull/3375/files#r1822708150